### PR TITLE
Adds AllowClobber powershell_package resource

### PR DESCRIPTION
**Pull request by mikef-nl**
## Description
Cuurently powershell_package does not allow for the usage of the -AllowClobber switch

## Related Issue
[Issue](https://github.com/chef/chef/issues/14381)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] If `Gemfile.lock` has changed, I have used `--conservative` to do it and included the full output in the Description above.
- [ ] All new and existing tests passed.
- [x] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

**Comment by tpowell-progress** [comment]
@mikef-nl Can we get a DCO on the commit, please?

File: lib/chef/resource/powershell_package.rb [hunk 1]
```diff
@@ -44,6 +44,10 @@ class PowershellPackage < Chef::Resource::Package
         description: "Skip validating module author.",
         default: false, introduced: "14.3", desired_state: false
 
+      property :allow_clobber,  [TrueClass, FalseClass],
+        description: "Overrides warning messages about installation conflicts about existing commands on a computer.",
+        default: false
```
Comment (tpowell-progress) [inline lib/chef/resource/powershell_package.rb:49]: @mikef-nl Can you also add `introduced: "18.5"` so the `property` similar to the above properties?

**Review by tpowell-progress — CHANGES_REQUESTED** [review]
Add "introduced" attribute to property.

File: lib/chef/resource/powershell_package.rb (hunk 1 above)
Comment (mikef-nl) [reply lib/chef/resource/powershell_package.rb]: Have added it as requested

**Comment by tpowell-progress** [comment]
@mikef-nl for simplicity of DCO, can you `git rebase -i origin/main` and squash all the commits into one and then `git commit --amend -s` to sign the commit and `git push -f`?

**Comment by tpowell-progress** [comment]
> @mikef-nl for simplicity of DCO, can you `git rebase -i origin/main` and squash all the commits into one and then `git commit --amend -s` to sign the commit and `git push -f`?

Bump for visibility. Looks like 3 of the commits are missing DCO otherwise.

**Comment by tpowell-progress** [comment]
@mikef-nl rerunning the suite

**Review by tpowell-progress — APPROVED** [review]

**Pull request merged**